src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -2087,6 +2087,7 @@ public synchronized void removeStaleRunningEntity(ADTask adTask, String entity,
     public boolean skipUpdateHCRealtimeTask(String detectorId, String error) {
         ADRealtimeTaskCache realtimeTaskCache = adTaskCacheManager.getRealtimeTaskCache(detectorId);
         return realtimeTaskCache != null
+            && realtimeTaskCache.getInitProgress() != null
```

Comment:
What was the purpose of this change?

Reply:
this is to prevent a null pointer exception I have seen:

```
[2021-08-17T17:32:26,571][ERROR][o.o.a.AnomalyDetectorJobRunner] [integTest-1] Failed to index anomaly result for MagnVXsBo0HEYi-dIN6x
java.lang.NullPointerException: Cannot invoke “java.lang.Float.floatValue()” because the return value of “org.opensearch.ad.task.ADRealtimeTaskCache.getInitProgress()” is null
        at org.opensearch.ad.task.ADTaskManager.skipUpdateHCRealtimeTask(ADTaskManager.java:2048) ~[opensearch-anomaly-detection-1.1.0.0-SNAPSHOT.jar:1.1.0.0-SNAPSHOT]
        at org.opensearch.ad.AnomalyDetectorJobRunner.updateRealtimeTask(AnomalyDetectorJobRunner.java:524) ~[opensearch-anomaly-detection-1.1.0.0-SNAPSHOT.jar:1.1.0.0-SNAPSHOT]
        at org.opensearch.ad.AnomalyDetectorJobRunner.indexAnomalyResult(AnomalyDetectorJobRunner.java:487) [opensearch-anomaly-detection-1.1.0.0-SNAPSHOT.jar:1.1.0.0-SNAPSHOT]
        at org.opensearch.ad.AnomalyDetectorJobRunner.lambda$runAdJob$3(AnomalyDetectorJobRunner.java:268) [opensearch-anomaly-detection-1.1.0.0-SNAPSHOT.jar:1.1.0.0-SNAPSHOT]
        at org.opensearch.action.ActionListener$1.onResponse(ActionListener.java:76) [opensearch-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
...

```